pkg/monitortests/kubeapiserver/apiunreachablefromclientmetrics/resolver.go

```diff
@@ -0,0 +1,81 @@
+package apiunreachablefromclientmetrics
+
+import (
+	"context"
+	"fmt"
+	"net"
+	"strings"
+
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes"
+)
+
+// TODO: move this to a reusable package
+func NewClusterInfoResolver(ctx context.Context, client kubernetes.Interface) (*clusterInfoResolver, error) {
+	kubeSvc, err := client.CoreV1().Services(metav1.NamespaceDefault).Get(ctx, "kubernetes", metav1.GetOptions{})
+	if err != nil {
+		return nil, fmt.Errorf("failed to retrieve cluster IP from kubernetes.default.svc - %v", err)
+	}
+
+	allNodes, err := client.CoreV1().Nodes().List(ctx, metav1.ListOptions{})
+	if err != nil {
+		return nil, fmt.Errorf("failed to retrieve nodes - %v", err)
+	}
+	if len(allNodes.Items) == 0 {
+		return nil, fmt.Errorf("unexpected empty list of nodes")
+	}
+	return &clusterInfoResolver{
+		serviceNetworkIP: kubeSvc.Spec.ClusterIP,
+		nodes:            allNodes.Items,
```

Comment:
Consider simply pre-computing the needed data at this point. e.g.:
```suggestion
                // ip -> name
		nodeNames: map[string]string
		// ip -> node-role & name -> node-role
		nodeRoles: map[string]string
```